sdcm/cluster.py

```diff
@@ -2766,6 +2784,9 @@ def node_setup(self, node, verbose=False, timeout=3600):
         :param verbose:
         """
         node.wait_ssh_up(verbose=verbose)
+        # update repo cache and system after system is up
+        node.update_repo_cache()
+        node.upgrade_system()
```

Comment:
why do we need to upgrade system on each node setup?
cc: @fruch

Reply:
We (SCT) only upgrade system to latest before installing scylla in the past, the upgrade isn't necessary. Actually the GCE images we used keep updating automatically.

I saw in recently upgrade jobs, db node setup is stable, but loader/monitor isn't stable, then I found the different. So I changed to upgrade system for db node/loader/monitor in node setup. It should be not the root reason of unstable.

If we think it's redundant, I'll test if it's stable after removing the upgrade for loader & monitor.

Reply:
@fruch WUT?

Reply:
keeping the all the package update to latest is a good practice
it can bite us at time, and not doing it might bite us too.

I think doing this mean we'll have to address any new issue related to 3rd party quicker, 
better sooner then later.

Reply:
- how long does it take to upgrade? 
- by upgrading always we lose the concept of the Image (AMI is an image, no?) 
- we will need to deal with timeouts, unavailable repos
are you sure that this is what we want?

Reply:
> keeping the all the package update to latest is a good practice
> it can bite us at time, and not doing it might bite us too.
> 
> I think doing this mean we'll have to address any new issue related to 3rd party quicker,
> better sooner then later.

You are right.

- If we reported some issue caused by old 3rd packages, it works well with latest 3rd packages, it will waste our time.
- If we find problem with latest 3rd packages, it's good, our users might touch same problem, we may need to provide them a workaround sometimes.

Reply:
> * how long does it take to upgrade?

Always very quick (less than 2 mins?) There is very small gap. Because:

- For GCE, use used a long term supported image, it's updated by gce automatically.
- For AWS (AMI), we actually updated the system (AMI) before installing scylla in preparing scylla AMI. https://github.com/scylladb/scylla/blob/master/dist/ami/files/scylla_install_ami#L64

Both GCE and AWS has their own internal mirror, so the download is very quick.

> * by upgrading always we lose the concept of the Image (AMI is an image, no?)

Good point.

- If the upgrade in SCT avoided some problem in AMI, we can't find this kind of issue. It's problem.
- If we don't upgrade in SCT, but our AMI user upgraded by themself and touched some problem. We won't catch problem for them.

In practice, the upgrade gap is very small, the effect is very small.

> * we will need to deal with timeouts, unavailable repos

We can add a timeout to avoid stuck. How about 10 mins?
unavailable repo will fail the command, but we have 3 retry.

>   are you sure that this is what we want?

I prefer the upgrade to catch the new issue.

Reply:
@benti what's your opinion?

Reply:
per our discussion, upgrade system should happen during upgrade test only, not for all SCT runs

Reply:
Updated PR to only upgrade system in upgrade test.